### ISSUE TITLE
Consider changing to LGPL (or anything less restrictive than GPL)

### ISSUE THREAD

**Issue by rnavarropiris** [slub/urnlib#12]
The GNU General Public License v3.0 imposes conditions on any software integrating a GPL license artifact which are rarely compatible with commercial software.

Switching to LGPL (GNU Lesser General Public License) would broaden the number of projects in which this library can be used.

**Comment by rnavarropiris** [comment]
I really want to use this library instead of having to re-code the same by myself, but the license is impeding me to do it...

**Comment by claussni** [comment]
Thanks for raising this issue. The licensing is twofold in this case. If you want to integrate or modify the source code, you need to conform to GPL. If you just want to use the binary artifact (as published on Maven Central) as is, an Apache license applies.

**Comment by claussni** [comment]
Maybe I can make the conditions more clear by removing the GPL source code license. The [Choose an open source license project](https://choosealicense.com/) provides a lot of useful information on that topic. I will give it some thought...

**Comment by rnavarropiris** [comment]
Oh, I see.

The binary being licensed under Apache solves my issue, maybe clarifying it in the README is enough.

 Thanks for the feedback! :)

**Comment by PyvesB** [comment]
I agree, it would definitely help to clarify and/or switch to a less restrictive license. 👍
